x-pack/plugins/alerting/server/lib/wrap_scoped_cluster_client.ts

```diff
@@ -71,6 +74,27 @@ export function createWrappedScopedClusterClientFactory(opts: WrapScopedClusterC
   };
 }
 
+export function wrapSearchSourceFetch(opts: WrapScopedClusterClientOpts) {
+  return async (searchSource: ISearchSource) => {
+    try {
+      const start = Date.now();
+      opts.logger.debug(
+        `executing query for rule ${opts.rule.alertTypeId}:${opts.rule.id} in space ${opts.rule.spaceId}`
+      );
+      const result = await searchSource.fetch({ abortSignal: opts.abortController.signal });
+
+      const durationMs = Date.now() - start;
+      opts.logMetricsFn({ esSearchDuration: result.took ?? 0, totalSearchDuration: durationMs });
+      return result;
+    } catch (e) {
+      if (opts.abortController.signal.aborted) {
+        throw new Error('Search has been aborted due to cancelled execution');
+      }
+      throw e;
```

Comment:
Just to gain knowledge as a start, when is a request aborted? When it's long running? So e.g. longer than the configured interval? If this is the case, given there's an always long running request, could this lead to an alert that is never "firing", since those requests never finish?

Reply:
Basically, rule task timeout is defined [here](https://github.com/elastic/kibana/blob/b84383e937e0106fc9346f94ddda29509d930c54/x-pack/plugins/alerting/server/lib/get_rule_task_timeout.ts#L12). When timeout is not configured, it will cancel request in `5m`. When requests will starts to be cancelled, it will lead to errors which user can see in `Error log` tab of created rule. @ymao1 Can user change rule task timeout? Will user be notified about errors in the rule?

Reply:
@Dmitriynj You are correct that the default timeout is `5m`. We have provided several ways of overriding this default:

- Rule types can set rule type specific timeouts using the `ruleTaskTimeout` field when registering a rule type: https://github.com/elastic/kibana/blob/123b231c60503e9988ffa69a66c8503d64db64f1/x-pack/plugins/alerting/common/rule_type.ts#L37
- Users can override both the default and rule type specific timeouts in the kibana config: https://github.com/elastic/kibana/pull/128294

> When requests will starts to be cancelled, it will lead to errors which user can see in Error log tab of created rule.

This is exactly correct. If a rule is timing out with every execution, users will see these timeouts and any errors that occurred during timeouts in the `Error log` tab of the rule details view.